src/module_specifier.rs

```diff
@@ -54,13 +55,27 @@ fn specifier_from_path(_path: PathBuf) -> ModuleSpecifier {
 pub fn resolve_import(
   specifier: &str,
   referrer: &ModuleSpecifier,
+  maybe_npm_resolver: Option<&dyn NpmResolver>,
 ) -> Result<ModuleSpecifier, SpecifierError> {
   let url = match ModuleSpecifier::parse(specifier) {
     // 1. Apply the URL parser to specifier.
-    //    If the result is not failure, return he result.
+    //    If the result is not failure, return the result.
     Ok(url) => url,
 
-    // 2. If specifier does not start with the character U+002F SOLIDUS (/),
+    // 2. If specifier is a node specifier without node: scheme, return
+    //    the specifier with node: scheme added.
+    Err(ParseError::RelativeUrlWithoutBase)
+      if maybe_npm_resolver.map_or(false, |npm_resolver| {
+        npm_resolver.is_builtin_node_module_name(specifier)
```

Comment:
Perhaps instead we could use:

```rs
// probably doesn't compile, but this is the basic idea
ModuleSpecifier::parse(format!("node:{}", specifier)).ok().and_then(|specifier| npm_resolver.resolve_builtin_node_module(&specifier).ok().flatten()).is_some()
```

Then we wouldn't need a separate `is_builtin_node_module_name` function

Reply:
Thanks! Removed `is_builtin_node_module_name`